svm/src/message_processor.rs

```diff
@@ -86,7 +87,87 @@ pub(crate) fn process_message(
             .total_us += process_instruction_us;
 
         result.map_err(|err| {
-            TransactionError::InstructionError(top_level_instruction_index as u8, err)
+            let transaction_context: &TransactionContext = invoke_context.transaction_context;
+            let responsible_program_account_index = &transaction_context
+                // By definition the last instruction (outer or inner) in the trace before the trace
+                // stopped being appended to is the one that encountered an error.
+                .get_instruction_trace_length()
+                .checked_sub(1)
+                .and_then(|index_in_trace| {
+                    transaction_context
+                        .get_instruction_context_at_index_in_trace(index_in_trace)
+                        .ok()
+                })
```

Comment:
This is my lack of understanding, but would it be simpler to call `get_current_instruction_context`? On the flip side, it looks like it boils down to almost exactly the same code